samples/bluetooth/encrypted_advertising/peripheral/src/peripheral_ead.c

```diff
@@ -0,0 +1,377 @@
+
+	if (!err) {
+		LOG_DBG("Security changed: %s level %u", addr, level);
+	} else {
+		LOG_DBG("Security failed: %s level %u err %d", addr, level, err);
+	}
+}
+
+static void auth_passkey_confirm(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	char passkey_str[7];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	snprintk(passkey_str, 7, "%06u", passkey);
+
+	LOG_DBG("Confirm passkey for %s: %s", addr, passkey_str);
+
+	k_poll_signal_raise(&passkey_enter_signal, 0);
+}
+
+static void auth_passkey_display(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	char passkey_str[7];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	snprintk(passkey_str, 7, "%06u", passkey);
+
+	LOG_DBG("Passkey for %s: %s", addr, passkey_str);
+}
+
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("Pairing cancelled: %s", addr);
+}
+
+static int update_security(void)
+{
+	int err;
+
+	await_signal(&passkey_enter_signal);
+
+	err = bt_conn_auth_passkey_confirm(default_conn);
+	if (err) {
+		LOG_DBG("Failed to confirm passkey.");
+		return -1;
+	}
+
+	return 0;
+}
+
+static int init_bt(void)
+{
+	int err;
+
+	k_poll_signal_init(&disconn_signal);
+	k_poll_signal_init(&passkey_enter_signal);
+
+	err = bt_enable(NULL);
+	if (err) {
+		LOG_ERR("Bluetooth init failed (err %d)", err);
+		return -1;
+	}
+
+	LOG_DBG("Bluetooth initialized");
+
+	err = bt_unpair(BT_ID_DEFAULT, BT_ADDR_LE_ANY);
+	if (err) {
+		LOG_ERR("Unpairing failed (err %d)", err);
+	}
+
+	peripheral_cb.connected = connected;
+	peripheral_cb.disconnected = disconnected;
+	peripheral_cb.security_changed = security_changed;
+
+	bt_conn_cb_register(&peripheral_cb);
+
+	peripheral_auth_cb.pairing_confirm = NULL;
+	peripheral_auth_cb.passkey_confirm = auth_passkey_confirm;
+	peripheral_auth_cb.passkey_display = auth_passkey_display;
+	peripheral_auth_cb.passkey_entry = NULL;
+	peripheral_auth_cb.oob_data_request = NULL;
+	peripheral_auth_cb.cancel = auth_cancel;
+
+	err = bt_conn_auth_cb_register(&peripheral_auth_cb);
+	if (err) {
+		LOG_ERR("Failed to register bt_conn_auth_cb (err %d)", err);
+		return -1;
+	}
+
+	return 0;
+}
+
+int run_peripheral_sample(void)
+{
+	int err;
+	struct bt_le_ext_adv *adv = NULL;
+
+	err = init_bt();
+	if (err) {
+		return -1;
+	}
+
+	/* Setup advertiser */
+	err = create_adv(&adv);
+	if (err) {
+		return -1;
+	}
+	err = start_adv(adv);
+	if (err) {
+		return -1;
+	}
+	err = set_ad_data(adv);
+	if (err) {
+		return -1;
+	}
+
+	/* Wait for the peer to update security */
+	err = update_security();
+	if (err) {
+		LOG_ERR("Failure during security update");
+		return -1;
+	}
+
+	/* Wait for the peer to disconnect */
+	await_signal(&disconn_signal);
+
+	/* Restart advertising */
+	err = start_adv(adv);
+	if (err) {
+		return -1;
+	}
+	err = set_ad_data(adv);
```

Comment:
For the sake of readability, please keep an empty line after closing brackets. 
```suggestion
	}
	
	err = set_ad_data(adv);
```